pointcloud/basic_octree.go

```diff
@@ -1,92 +1,94 @@
-package octree
+package pointcloud
 
 import (
-	"context"
-
-	"github.com/edaniels/golog"
 	"github.com/golang/geo/r3"
 	"github.com/pkg/errors"
-
-	pc "go.viam.com/rdk/pointcloud"
 )
 
 const (
+	internalNode = NodeType(iota)
+	leafNodeEmpty
+	leafNodeFilled
+	octreeVersion = 1.0
 	// This value allows for high level of granularity in the octree while still allowing for fast access times
 	// even on a pi.
 	maxRecursionDepth = 1000
 )
 
-// basicOctree is a data structure that represents a basic octree structure with information regarding center
-// point, side length and node data.
-type basicOctree struct {
-	logger     golog.Logger
+// NodeType represents the possible types of nodes in an octree.
+type NodeType uint8
+
+// BasicOctree is a data structure that represents a basic octree structure with information regarding center
+// point, side length and node data. An octree is a data structure that recursively partitions 3D space into
+// octants to represent occupancy. It is a storage format for a pointcloud that allows for better searchability
+// and serialization.
+type BasicOctree struct {
 	node       basicOctreeNode
 	center     r3.Vector
 	sideLength float64
 	size       int
-	meta       pc.MetaData
+	meta       MetaData
 }
 
 // basicOctreeNode is a struct comprised of the type of node, children nodes (should they exist) and the pointcloud's
 // PointAndData datatype representing a point in space.
 type basicOctreeNode struct {
 	nodeType NodeType
-	children []*basicOctree
-	point    pc.PointAndData
+	children []*BasicOctree
+	point    PointAndData
 }
 
-// New creates a new basic octree with specified center, side and metadata.
-func New(ctx context.Context, center r3.Vector, sideLength float64, logger golog.Logger) (Octree, error) {
+// NewBasicOctree creates a new basic octree with specified center, side and metadata.
+func NewBasicOctree(center r3.Vector, sideLength float64) (PointCloud, error) {
```

Comment:
I would expect any function which has `Octree` in the name, to have a function signature that takes or returns a `BasicOctree` or `*BasicOctree`. 

I expect this as `PointCloud` is supposed to be in memory representation agnostic (correct me if I'm wrong) so saying that it returns a `PointCloud` interface type, which in reality will always be a `BasicOctree` concrete type, seems like being overly generic (which can cause confusion).

What is the benefit of returning the interface BasicOctree implements w/o being specific about the only concrete type this (and other `*Octree` functions) return?